### PR TITLE
Use NumberInputLazy for stat constraints/bonus

### PR DESCRIPTION
**Pull request by frzyc**
## Describe your changes

Update Bonus stast and Stat Constraints to the new NumberInputLazy element.

## Issue or discord link

- 

## Testing/validation

image
Screenshot 2024-09-25 200922

## Checklist before requesting a review (leave this PR as draft if any part of this list is not done.)

- [ ] I have commented my code in hard-to understand areas.
- [ ] I have made corresponding changes to README or wiki.
- [ ] For front-end changes, I have updated the corresponding English translations.
- [ ] I have run `yarn run mini-ci` locally to validate format and lint.
- [ ] If I have added a new library or app, I have updated the deployment scripts to ignore changes as needed

### CODE REVIEW

**Comment by nguyentvan7** [comment]
Do we need to update tc tab too?

**Comment by frzyc** [comment]
> Do we need to update tc tab too?

TC constraints are being updated in https://github.com/frzyc/genshin-optimizer/pull/2460

File: libs/gi/page-team/src/CharacterDisplay/Tabs/TabOptimize/Components/OptimizationTargetEditorList.tsx [hunk 1]
```diff
@@ -165,39 +167,41 @@ function OptimizationTargetEditorItem({
...
+              >
+                <DeleteForever fontSize="small" />
+              </IconButton>
+            </InputAdornment>
+          ),
         }}
       />
-      <CustomNumberInputButtonGroupWrapper sx={{ flexBasis: 150, flexGrow: 1 }}>
-        <CustomNumberInput
-          float
```
Comment (nguyentvan7) [inline libs/gi/page-team/src/CharacterDisplay/Tabs/TabOptimize/Components/OptimizationTargetEditorList.tsx:181]: Lost the ability to enter floats

**Comment by frzyc** [comment]
Also probably need to extend the changes to SRO, but probably in a later PR (SRO has min and max so needs more cooking)

**Review by nguyentvan7 — APPROVED** [review]

**Pull request merged**